File: ios/JJMproject/Classes/JSBridge/Modules/File/SandboxModule.m
//
//  NetworkModule.m
//  MarioFramework
//
//  Created by MeePwn on 2017/11/15.
//  Copyright © 2017年 Facebook. All rights reserved.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(SandboxModule, NSObject)

RCT_EXTERN_METHOD(documentPath:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(cachePath:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject)

@end

File: ios/JJMproject/Classes/JSBridge/Modules/File/SandboxModule.swift
//
//  FileModule.swift
//  MarioFramework
//
//  Created by MeePwn on 2017/11/29.
//  Copyright © 2017年 Facebook. All rights reserved.
//

import UIKit

@objc(SandboxModule)
class SandboxModule: NSObject {
  
  private lazy var fileUtility: FileUtility = {
    return FileUtility()
  }()
  
  @objc func documentPath(_ resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    resolve(fileUtility.documentPath())
  }
  
  @objc func cachePath(_ resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    resolve(fileUtility.cachePath())
  }
  
}
